Core/Application/CQRS/Soils/InsertSoilLayerTraitsCommand.cs

```diff
@@ -45,10 +45,8 @@ private Unit Handler(InsertSoilLayerTraitsCommand request)
 
             // All the non-key properties of an entity
             var layer_props = _context.GetEntityProperties(typeof(SoilLayer));
-            IEntity layer = null;
-            Func<IEntity, bool> layer_matches = other =>
-                    layer_props.All(i => i.GetValue(layer)?.ToString() == i.GetValue(other)?.ToString());
 
+            IEntity layer = null;
```

Comment:
Why not declare this as a SoilLayer?

Reply:
Convenience mostly, parts of that method were copy/pasted from other import functions. I could change it if necessary but I don't think it really makes a difference. 

Though looking at it again it doesn't need to be declared outside the loop any more. Originally it was part of a local function which required it to be instantiated, but that's no longer the case.

Reply:
Cleaned it up in 42594e8